Src/PCompiler/CompilerCore/TypeChecker/Scope.cs

```diff
@@ -32,9 +34,11 @@ public class Scope
         private readonly IDictionary<string, TypeDef> typedefs = new Dictionary<string, TypeDef>();
         private readonly IDictionary<string, Variable> variables = new Dictionary<string, Variable>();
 
-        private Scope(ITranslationErrorHandler handler, Scope parent = null)
+        private Scope(ITranslationErrorHandler handler, ILocationResolver locationResolver, IList<string> projectDependencies, Scope parent)
```

Comment:
make `parent = null` so that its not a required argument and then drop the `null`

Reply:
Done via 6758a4af0c3e39ef87622405bb8c78c12dea7a76